src/Physics/HadronTransport/INukeUtils2025.cxx

```diff
@@ -0,0 +1,1934 @@
+      P3tL = 0;
+      P3zL = .001;
+      P3L = .001;
+      E3L = TMath::Sqrt(P3L*P3L + M3*M3);
+    }
+
+  tP3L = P3zL*tbetadir + P3tL*tTrans;
+  tPiL = PizL*tbetadir + PitL*tTrans;
+  tP3L.Rotate(phi3,tbetadir);
+  tPiL.Rotate(phi3,tbetadir);
+
+  // second sequence, handle formally composite particles 4 and 5
+  tbeta2 = tPiL * (1.0 / EiL);
+  tbetadir2 = tbeta2.Unit();
+  beta2 = tbeta2.Mag();
+  gm2 = 1.0 / TMath::Sqrt(1.0 - beta2*beta2);
+
+  E4CM2 = (M*M + M4*M4 - M5*M5) / (2*M);
+  E5CM2 = M - E4CM2;
+  P4CM2 = TMath::Sqrt(E4CM2*E4CM2 - M4*M4);
+
+  tVect.SetXYZ(1,0,0);
+  if(TMath::Abs((tVect - tbetadir2).Mag())<.01) tVect.SetXYZ(0,1,0);
+  theta5 = tVect.Angle(tbetadir2);
+  tTrans2 = (tVect - TMath::Cos(theta5)*tbetadir2).Unit();
+
+  P4zL = gm2*beta2*E4CM2 + gm2*P4CM2*TMath::Cos(theta4);
+  P4tL = P4CM2*TMath::Sin(theta4);
+  P5zL = gm2*beta2*E5CM2 - gm2*P4CM2*TMath::Cos(theta4);
+  P5tL = - P4tL;
+
+  P4L = TMath::Sqrt(P4zL*P4zL + P4tL*P4tL);
+  P5L = TMath::Sqrt(P5zL*P5zL + P5tL*P5tL);
+  E4L = TMath::Sqrt(P4L*P4L + M4*M4);
+  E5L = TMath::Sqrt(P5L*P5L + M5*M5);
+
+  // handle very low momentum particles
+  if(!(TMath::Finite(P4L)) || P4L < .001)
+    {
+      LOG("INukeUtils",pINFO)
+        << "Particle 4 " << M4 << " momentum small or non-finite: " << P4L
+        << "\n" << "--> Assigning .001 as new momentum";
+      P4tL = 0;
+      P4zL = .001;
+      P4L = .001;
+      E4L = TMath::Sqrt(P4L*P4L + M4*M4);
+    }
+  if(!(TMath::Finite(P5L)) || P5L < .001)
+    {
+      LOG("INukeUtils",pINFO)
+        << "Particle 5 " << M5 << " momentum small or non-finite: " << P5L
+        << "\n" << "--> Assigning .001 as new momentum";
+      P5tL = 0;
+      P5zL = .001;
+      P5L = .001;
+      E5L = TMath::Sqrt(P5L*P5L + M5*M5);
+    }
+
+  tP4L = P4zL*tbetadir2 + P4tL*tTrans2;
+  tP5L = P5zL*tbetadir2 + P5tL*tTrans2;
+  tP4L.Rotate(phi4,tbetadir2);
+  tP5L.Rotate(phi4,tbetadir2);
+
+  // pauli blocking
+  if(P3L < FermiMomentum || ( pdg::IsNeutronOrProton(s2->Pdg()) && P4L < FermiMomentum ) )
+  {
+    LOG("INukeUtils",pNOTICE)
+      << "PionProduction fails because of Pauli blocking - retry kinematics";
+    exceptions::INukeException exception;
+    exception.SetReason("PionProduction final state not determined");
+    throw exception;
+    return false;
+  }
+
+  // create scattered particles w/ appropriate momenta, code, and status
+  // Set moms to be the moms of the hadron that was cloned
+  s1->SetFirstMother(p->FirstMother());
+  s2->SetFirstMother(p->FirstMother());
+  s3->SetFirstMother(p->FirstMother());
+  s1->SetLastMother(p->LastMother());
+  s2->SetLastMother(p->LastMother());
+  s3->SetLastMother(p->LastMother());
+
+  TLorentzVector(tP3L,E3L);
+  TLorentzVector(tP4L,E4L);
+  TLorentzVector(tP5L,E5L);
+
+  s1->SetMomentum(TLorentzVector(tP3L,E3L));
+  s2->SetMomentum(TLorentzVector(tP4L,E4L));
+  s3->SetMomentum(TLorentzVector(tP5L,E5L));
+  int mode = kIMdHA;
```

Comment:
Setting `mode` here to a fixed value and then following it immediately with an if/then/else that always takes the "else" path is odd.